src/measure_theory/indicator_function.lean

```diff
@@ -0,0 +1,233 @@
+/-
+Copyright (c) 2020 Zhouhang Zhou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Zhouhang Zhou
+-/
+
+import measure_theory.measure_space
+import analysis.normed_space.basic
+
+/-!
+# Indicator function
+
+`indicator (s : set α) (f : α → β) (a : α)` is `f a` if `a ∈ s` and is `0` otherwise.
+
+## Implementation note
+
+In mathematics, an indicator function or a characteristic function is a function used to indicate
+membership of an element in a set `s`, having the value `1` for all elements of `s` and the value  `0`
+otherwise. But since it is usually used to restrict a function to a certain set `s`, we let the
+indicator function take the value `f x` for some function `f`, instead of `1`. If the usual indicator
+function is needed, just set `f` to be the constant function `λx, 1`.
+
+## Tags
+indicator, characteristic
+-/
+
+noncomputable theory
+open_locale classical
+
+open set measure_theory filter
+
+universes u v
+variables {α : Type u} {β : Type v}
+
+section has_zero
+variables [has_zero β] {s t : set α} {f g : α → β} {a : α}
+
+/-- `indicator s f a` is `f a` if `a ∈ s`, `0` otherwise.  -/
+@[reducible]
+def indicator (s : set α) (f : α → β) : α → β := λ x, if x ∈ s then f x else 0
```

Comment:
Probably the definition of  `indicator` and basic properties (i.e., properties that don't need a measure) should be in `data/`, not in `measure_theory`.

Reply:
It's done but I'm not sure I like it. I think there is nothing interesting in the basic properties of the indicator function. People can just use if-then-else.